File: api/MfaApi/src/Modules/Membership/Contracts/GetMembershipsRequest.cs
using MfaApi.Core.Sort;
using MfaApi.Core.Pagination;

namespace MfaApi.Modules.Membership;

public class GetMembershipsRequest: PaginationRequest {
    public int YearPaid { get; set; } = DateTime.UtcNow.Year;
    public string? Query { get; set; }
    public MembershipType? MembershipType { get; set; }
    public SortOrder? SortStartDate { get; set; }
    public bool? HasPaid { get; set; }
    public DateTime? SinceFrom { get; set; }
    public DateTime? SinceTo { get; set; }
    public bool? IsActive { get; set; }
}

File: dashboard/src/modules/membership/types/get-memberships-request.ts
import { SortOrder } from '@/types/sort-order';

import { MembershipType } from './membership-type';

export interface GetMembershipsRequest {
  yearPaid: number;
  query: string | null;
  membershipType: MembershipType | null;
  sortStartDate: SortOrder | null;
}